pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1649,6 +1649,52 @@ func computeGrowDistribution(groups []jsonManagedInstanceGroup, newNodeCount int
 	return addCount
 }
 
+// Shrink shrinks the cluster by deleting the given VMs. This is only supported
+// for managed instance groups. Currently, nodes should only be deleted from the
+// tail of the cluster, because this function does not yet support renaming
+// nodes and re-syncing DNS names to keep the cluster in a consistent state.
+func (p *Provider) Shrink(l *logger.Logger, vmsToDelete vm.List, clusterName string) error {
+	if !isManaged(vmsToDelete) {
+		return errors.New("shrinking is only supported for managed instance groups")
+	}
+
+	project := vmsToDelete[0].Project
+	groupName := instanceGroupName(clusterName)
+	groups, err := listManagedInstanceGroups(project, groupName)
+	if err != nil {
+		return err
+	}
+	zones := make(map[string]struct{})
+	for _, group := range groups {
+		zones[group.Zone] = struct{}{}
+	}
+
+	g := errgroup.Group{}
+	for _, cVM := range vmsToDelete {
```

Comment:
Do we have to worry about API limits? i.e., number of concurrent `delete-instances`

Reply:
Yep, we could add a limit to the `errgroup`, or as you mentioned pass multiple instances per group. I opt for the latter, will update.